psi4/driver/procrouting/dft_funcs/dict_gga_funcs.py

```diff
@@ -0,0 +1,194 @@
+    "c_functionals": {"GGA_C_PBE": {}},
+    "citation": '    J.P. Perdew et. al., Phys. Rev. Lett., 77(18), 3865-3868, 1996\n',
+    "description": '    PBE GGA Exchange-Correlation Functional\n',
+}
+
+pw91 = {
+    "name": "PW91",
+    "x_functionals": {"GGA_X_PW91": {}},
+    "c_functionals": {"GGA_C_Pw91": {}},
+    "citation": '    J.P. Perdew et. al., Phys. Rev. B., 46(11), 6671-6687, 1992\n',
+    "description": '    PW91 GGA Exchange-Correlation Functional\n',
+}
+
+mpwpw = {
+    "name": "MPWPW",
+    "x_functionals": {"GGA_X_mPW91": {}},
+    "c_functionals": {"GGA_C_Pw91": {}},
+    "citation": '    C. Adamo, V. Barone, J. Chem. Phys., 108, 664, 1998\n',
+    "description": '    mPWPW GGA Exchange-Correlation Functional\n',
+}
+
+bp86 = {
+    "name": "BP86",
+    "x_functionals": {"GGA_X_B88": {}},
+    "c_functionals": {"GGA_C_P86": {}},
+    "citation": '   A. D. Becke, Phys. Rev. A, 38, 3098-3100, 1988\n   J. P. Perdew, Phys. Rev. B, 33, 8822, 1986\n',
+    "description": '    BP86 GGA Exchange-Correlation Functional\n',
+}
+
+ft97 = {
+    "name": "FT97",
+    "x_functionals": {"GGA_X_FT97_B": {}},
+    "c_functionals": {"GGA_C_FT97": {}},
+    "citation": '    M. Filatov and W. Theil, Int. J. Quant. Chem., 62, 603-616, 1997\n',
+    "description": '   FT97 GGA Exchange-Correlation Functional\n',
+}
+
+sogga = {
+    "name": "SOGGA",
+    "x_functionals": {"GGA_X_SOGGA": {}},
+    "c_functionals": {"GGA_C_PBE": {}},
+    "citation": '    Y. Zhao, and D. G. Truhlar, J. Chem. Phys. 128, 184109, 2008\n' +
+                '    J. Perdew, K. Burke, M. Ernzerhof, Phys. Rev. Lett. 77, 3865-3868, 1996\n',
+    "description": '   SOGGA Exchange + PBE Correlation Functional\n',
+}
+
+sogga11 = {
+    "name": "SOGGA11",
+    "x_functionals": {"GGA_X_SOGGA11": {}},
+    "c_functionals": {"GGA_C_SOGGA11": {}},
+    "citation": '    R. Peverati, Y. Zhao, and D. G. Truhlar, J. Phys. Chem. Lett. 2, 1991-1997, 2011\n',
+    "description": '   SOGGA11 Exchange-Correlation Functional\n',
+}
+
+
+functional_list = {
+    "TEST-B86BPBE": b86bpbe,
+    "TEST-BLYP": blyp,
+    "TEST-BLYP-D2": blyp_d2,
+    "TEST-BLYP-D3": blyp_d3,
+    "TEST-BLYP-D3BJ": blyp_d3bj,
+    "TEST-BLYP-D3M": blyp_d3m,
+    "TEST-BLYP-D3MBJ": blyp_d3mbj,
+    "TEST-PW86PBE": pw86pbe,
+    "TEST-PBE": pbe,
+    "TEST-BP86": bp86,
+    "TEST-PW91": pw91,
+    "TEST-FT97": ft97,
+    "TEST-BOP": bop,
+    "TEST-MPWPW": mpwpw,
+    "TEST-SOGGA11": sogga11,
+    "TEST-SOGGA": sogga,
+    "TEST-OPBE-D"    : {"name": "OPBE-D"   , "xc_functionals": {"GGA_XC_OPBE_D"   : {}}, "dispersion": {"type": "d3zero", "params": {'s6': 1.0,  's8': 1.494, 'sr6': 1.128, 'alpha6': 14.0}}},
+    #"TEST-OPWLYP_D"  : {"name": "OPWLYP-D" , "xc_functionals": {"GGA_XC_OPWLYP_D" : {}}}, # no dispersion
+    #"TEST-OBLYP_D"   : {"name": "OBLYP-D"  , "xc_functionals": {"GGA_XC_OBLYP_D"  : {}}}, # no dispersion
+    "TEST-HCTH407P" : {"name": "HCTH/407P", "xc_functionals": {"GGA_XC_HCTH_407P": {}}},
```

Comment:
Might be nice to automatically build the `xc_functionals` list as well like we do in `xc_funcs.py`.

Reply:
Is there a reason why you want to keep `xc_functionals` separate from the others? I basically just went through the `xc_funcs_list` and split them across the other files, so that when one looks for a hybrid there's only one place where it's defined, not two.

Reply:
Mostly because the XC list is either copied as a single block from LibXC itself or in the future we will query the library to see what XC funcs it has. Keeping it all in separate files will make it hard to figure out where new functionals go as LibXC adds new functionals at quite a ferocious rate to keep up with literature.